modules/services/xidlehook.nix

```
@@ -0,0 +1,144 @@
+      type = types.attrsOf types.str;
+      default = { };
+      example = literalExample ''
+        {
+          "primary-display" = "$(xrandr | awk '/ primary/{print $1}')";
+        }
+      '';
+      description = ''
+        Extra environment variables to be exported in the script.
+        These options are passed unescaped as <code>export NAME=value</code>.
+      '';
+    };
+
+    not-when-fullscreen = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = "Disable locking when a fullscreen application is in use.";
+    };
+
+    not-when-audio = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = "Disable locking when audio is playing.";
+    };
+
+    once = mkEnableOption "running the program once and exiting";
+
+    timers = mkOption {
+      type = types.listOf (types.submodule {
+        options = {
+          delay = mkOption {
+            type = types.ints.unsigned;
+            example = 60;
+            description = "Time before executing the command.";
+          };
+          command = mkOption {
+            type = types.nullOr types.str;
+            example = literalExample ''
+              ''${pkgs.libnotify}/bin/notify-send "Idle" "Sleeping in 1 minute"
+            '';
+            description = "Command executed after the idle timeout is reached.";
+          };
+          canceller = mkOption {
+            type = types.str;
+            default = "";
+            example = literalExample ''
+              ''${pkgs.libnotify}/bin/notify-send "Idle" "Resuming activity"
+            '';
+            description = ''
+              Command executed when the user becomes active again.
+              This is only executed if the next timer has not been reached.
+            '';
+          };
+        };
+      });
+      default = [ ];
+      example = literalExample ''
+        [
+          {
+            delay = 60;
+            command = "xrandr --output \"$PRIMARY_DISPLAY\" --brightness .1";
+            canceller = "xrandr --output \"$PRIMARY_DISPLAY\" --brightness 1";
+          }
+          {
+            delay = 120;
+            command = "''${pkgs.writeScript "my-script" '''
+              # A complex script to run
+            '''}";
+          }
+        ]
+      '';
+      description = ''
+        A set of commands to be executed after a specific idle timeout.
+        The commands specified in <code>command</code> and <code>canceller</code>
+        are passed escaped to the script.
+        To use or re-use environment variables that are script-dependent, specify them
+        in the <code>environment</code> section.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.user.services.xidlehook = {
+      Unit = {
+        Description = "xidlehook service";
+        PartOf = [ "graphical-session.target" ];
+        After = [ "graphical-session.target" ];
```

```suggestion
        Description = "xidlehook service";
        PartOf = [ "graphical-session.target" ];
        After = [ "graphical-session.target" ];
        ConditionEnvironment = [ "DISPLAY" ];
```

For mixed X11 & Wayland setup, this will only start the services if the DISPLAY environment variable is set.

This needs to be tested on non-NixOS system since NixOS' xserver module always imports the DISPLAY environment variable for user services.

---

I added the additional flag, but I cannot test on a non-NixOS system. Not sure how to proceed.